### DISCUSSION TITLE
How to export enums with a capacitor plugin that is functional with jest? (capacitorExports problem)

### DISCUSSION THREAD

**Discussion by distante** [ionic-team/capacitor#3971] [Q&A]
I am working in a capacitor plugin and I have troubles with non-interfaces exports.

I am declaring and exporting an `enum` declaration in `definition.ts` containing the listeners the plugin can have.

The problem I am having is that in tests with `jest`  I get this error when I try to import one of the enums:

> ReferenceError: capacitorExports is not defined

Everything works as expected in the ionic/angular project it self but I am not able to run unit tests.

If I remove the `enum`s and use plain strings everything works but I do not want to force the consumer to use magic strings.

another problem I have is that when I want to enable the web version importing the plugin as [recommended in the docs](https://capacitorjs.com/docs/plugins/web) sends the same error.

```
// THIS BREAKS TESTS

// Import custom plugin package for web support too
import 'my-plugin';
```

**Comment by pwespi** [comment]
I have a similar issue with my plugin: https://github.com/capacitor-community/bluetooth-le/issues/12

@distante Have you found any solutions/workaround in the meantime?

**Reply by distante** [reply]
@pwespi for the web version [I am using dynamic imports in the plugin consumer part](https://www.saninnsalas.com/referenceerror-capacitorexports-is-not-defined-while-running-jest-tests-in-a-project-with-capacitorjs-fix/)

And regarding the enums, I put them on another file so I can import them separately.

It is not pretty but at least I can keep testing my project.

**Reply by pwespi** [reply]
Thanks for the information.

I found that a commonjs output might help. But I'm not sure if it's the best solution. See https://github.com/capacitor-community/bluetooth-le/pull/14
